python_modules/dagster/dagster/_core/storage/upath_io_manager.py

```diff
@@ -191,33 +201,82 @@ def _load_multiple_inputs(self, context: InputContext) -> Dict[str, Any]:
             else False
         )
 
-        objs: Dict[str, Any] = {}
-        paths = self._get_paths_for_partitions(context)
-        backcompat_paths = self._get_multipartition_backcompat_paths(context)
+        paths = self._get_paths_for_partitions(context)  # paths for normal partitions
+        backcompat_paths = self._get_multipartition_backcompat_paths(
+            context
+        )  # paths for multipartitions
 
         context.log.debug(f"Loading {len(paths)} partitions...")
 
-        for partition_key, path in paths.items():
-            context.log.debug(f"Loading partition from {path} using {self.__class__.__name__}")
+        def load_partition(partition_key: Union[str, MultiPartitionKey]):
+            path = paths[partition_key]
+            backcompat_path = backcompat_paths.get(partition_key)
+
+            # 1. We will first try to load the partition from the normal path.
+            # 2. If it is not found, we will try to load it from the backcompat path.
+            # 3. If allow_missing_partitions is True, we will skip the partition if it was not found in any of the paths.
+            # otherwise, we will raise an error.
+
+            missing_partition_message = f"Couldn't load partition {partition_key} and skipped it "
+            "because the input metadata includes allow_missing_partitions=True"
+
             try:
+                context.log.debug(f"Loading partition from {path}...")
```

Comment:
Do we not want to include `self.__class__.__name__` in the debug message anymore?